libnetwork/osl/sandbox_linux_test.go

```diff
@@ -38,15 +38,18 @@ func generateRandomName(prefix string, size int) (string, error) {
 }
 
 func newKey(t *testing.T) (string, error) {
+	t.Helper()
 	name, err := generateRandomName("netns", 12)
 	if err != nil {
 		return "", err
 	}
 
 	name = filepath.Join("/tmp", name)
-	if _, err := os.Create(name); err != nil {
+	f, err := os.Create(name)
+	if err != nil {
 		return "", err
 	}
+	t.Cleanup(func() { _ = f.Close() })
```

Comment:
Why not close immediately?

Reply:
Oh man, LOL, need to look back at this; it's been a long time. ISTR there were a couple of tests that depended on a file having to be in use, but I may misremember. I'll have a look

Reply:
Alright, no idea why I did it; changed it to close immediately, and verified that the tests were still happy